### ISSUE TITLE
Incorrect app detection

### ISSUE THREAD

**Issue by lawson85** [0x192/Universal-Android-Debloater#86]
**Describe the bug**
All apps installed on my Samsung Galaxy core Prime are categorised as unlisted. Even com.android.chrome, which is in your lists as Google list. Plugging my other phones in - Honor 10 lite and Sony X work correctly. I suspect it has to do with Android 5.1.1 which my Galaxy is. 

**Expected behavior**
I expected to see my apps in the correct lists and at least one to be in the recommended which is empty at the moment. 

**You have a solution?**
I should buy a bloatfree device

**UAD log**
No log was generated

**Comment by 0x192** [comment]
>  I should buy a bloatfree device

I must admit this made me laugh!

@lawson85 & @MPeti1

My only guess on what going on is that your phones returns strange invisible characters at the end of each line when UAD tries to fetch the packages (`adb shell pm list packages -s`). Yet I've already made sure to strip them by precaution.

I'll give you a special debug build to test things out . Keep tuned.

**Comment by 0x192** [comment]
[Special windows build](https://github.com/0x192/universal-android-debloater/suites/4886801289/artifacts/139992090)

The issue may be fixed. If not please upload the logfile which is now located in a more standard location `C:\Users\<User>\AppData\Local\uad`

**Comment by MPeti1** [comment]
Hmm, running `adb shell pm list packages -s > adb-packages` on WSL makes a file that only has CRLF at the end:
![image](https://user-images.githubusercontent.com/9215491/148808495-5541a146-e4d3-4656-ae87-c1ccd4e5df74.png)

**Comment by 0x192** [comment]
> Hmm, running adb shell pm list packages -s > adb-packages on WSL makes a file that only has CRLF at the end:

Is there a reason you run it on WSL and not on Windows directly?

**Comment by MPeti1** [comment]
I hate the Windows CMD because it's unusable, but that's all. I already had a WSL tab open in Windows Terminal.

**Comment by MPeti1** [comment]
But I still started the adb-server that is installed on Windows, because it's own doesn't work in WSL 1

**Comment by MPeti1** [comment]
Where could I make the log file available to you privately?
Though I can still send an excerpt (certain lines) if that's helpful here, if that's also ok.

**Comment by 0x192** [comment]
> I hate the Windows CMD because it's unusable, but that's all. I already had a WSL tab open in Windows Terminal.

Yeah CMD is awful. The new [Windows Terminal](https://github.com/microsoft/terminal) or [Cmder](https://github.com/cmderdev/cmder) are way better.

There is no sensitive information in the logfile except maybe your Windows username (that you can remove if you want).

**Comment by MPeti1** [comment]
I treat my app list and phone model as senstitive. I think not much people had this type of phone, and also I have a lot of apps installed.

**Comment by 0x192** [comment]
FYI, this logfile only contains system apps (not the apps you installed) and I don't need all of them. I just want to know if there is some weird characters at the end of the lines.

**Comment by MPeti1** [comment]
Oh, ok. here is the file:
[UAD_20220110.log](https://github.com/0x192/universal-android-debloater/files/7841098/UAD_20220110.log)

**Comment by 0x192** [comment]
Oh there is no bug actually. All the packages marked as `Unlisted` by UAD aren't in the [uad_list.json](https://github.com/0x192/universal-android-debloater/blob/main/resources/assets/uad_lists.json).

That's the first time I see that but yeah there is no documentation for all of this packages.

You know what to do if you want to help :wink:

**Comment by MPeti1** [comment]
Oh, I see.

Well, there are quite a few packages that are not that hard to decipher their use, so I think I'll try to submit a list.

**Comment by MPeti1** [comment]
> Oh there is no bug actually

On the other hand, this does not really surprise me, by looking at the programming language of UAD :D

**Comment by lawson85** [comment]
UAD 0.5 has the same issue with my Samsung Galaxy Core prime and for some reason the special debug build link is not working : #197

**Comment by 0x192** [comment]
@ericzolf

This issue is really weird because there is no error. For example, `android` [is in the list ](https://github.com/0x192/universal-android-debloater/blob/main/resources/assets/uad_lists.json#L19443) but is not detected on your phone even if it is obviously there. 

The "good" news is that I know someone with a FairPhone 3 so it will be easier to understand what is going on when I have the phone in hand (next weekend).

**Comment by ericzolf** [comment]
If I can help let me know, Fedora 36, from the command line:

```
$ adb shell pm list packages -s | wc -l
241
```
but the UI shows only 10:
![image](https://user-images.githubusercontent.com/3105685/177043910-811866d2-745e-444e-b7e1-adac67c1d1b1.png)

And I had the same issue on a freshly reset Galaxy S10e, so perhaps it's an issue of some library/dependency on the PC, not an issue of the smartphone?

**Comment by ericzolf** [comment]
Doesn't seem indeed to be related to the smartphone: my Fairphone 4 which used to work properly, doesn't anymore. Same Fedora installation (different software versions at that time, roughly one month ago, might even have been Fedora 35), same version of your software (0.5).

The only dependency I can think of (rust isn't installed), but it didn't change since months, let me know if you have another idea:

```
$ rpm -qa | grep android
android-tools-31.0.2-3.fc36.x86_64
```

**Comment by 0x192** [comment]
I've pushed a debug build to see exactly what packages the function used to fetch packages gets. The pipeline is running. I let you know when it ends.

You'll just need to launch UAD and see (from the logs or directly from the terminal) if only 10 packages are listed or not.

**Comment by 0x192** [comment]
Here we go: https://github.com/0x192/universal-android-debloater/suites/7196832181/artifacts/287677097

**Comment by kudotuanminh** [comment]
I'm facing the same problem where all the packages are categorized as unlisted and have no description on a Samsung Galaxy J7 2016 (J7109), on both stock ROM and a custom ROM based on Android 6.0.1.

[adb-packages.txt](https://github.com/0x192/universal-android-debloater/files/9103203/adb-packages.txt)
![image](https://user-images.githubusercontent.com/25059254/178760193-bbb9a868-6799-4e1c-b16a-aa3015a1679d.png)
